wandb/wandb_agent.py

```diff
@@ -496,12 +496,28 @@ def run_agent(
 def agent(sweep_id, function=None, entity=None, project=None, count=None):
     """Generic agent entrypoint, used for CLI or jupyter.
 
+    Will run a function or program with configuration parameters specified
+        by server.
+
     Args:
         sweep_id (dict): Sweep ID generated by CLI or sweep API
-        function (func, optional): A function to call instead of the "program" specifed in the config
+        function (func, optional): A function to call instead of the "program"
+            specifed in the config.
         entity (str, optional): W&B Entity
         project (str, optional): W&B Project
         count (int, optional): the number of trials to run.
+
+    Examples:
+        Run a sample sweep over a function:
+        def train():
+            run = wandb.init()
```

Comment:
We should use a context here as well, i.e. `with wandb.init() as run:` so that `wandb.finish()` is automatically called.

Reply:
ok